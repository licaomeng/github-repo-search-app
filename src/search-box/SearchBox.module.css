.form {
    display: flex;
    align-items: center;
    margin: 10px;
}

.searchBox {
    --color-fg-default: #24292f;
    --color-canvas-default: #ffffff;
    --color-border-default: #d0d7de;
    --color-primer-shadow-inset: inset 0 1px 0 rgba(208, 215, 222, 0.2);
    width: -webkit-fill-available;
    margin: 10px;
    padding: 5px 12px;
    font-size: 14px;
    line-height: 20px;
    color: var(--color-fg-default);
    vertical-align: middle;
    background-color: var(--color-canvas-default);
    background-repeat: no-repeat;
    background-position: right 8px center;
    border: 1px solid var(--color-border-default);
    border-radius: 6px;
    box-shadow: var(--color-primer-shadow-inset);
    transition: 80ms cubic-bezier(0.33, 1, 0.68, 1);
    transition-property: color, background-color, box-shadow, border-color;
}

.btn {
    --color-btn-text: #24292f;
    --color-btn-bg: #f6f8fa;
    --color-btn-border: rgba(27, 31, 36, 0.15);
    --color-btn-shadow: 0 1px 0 rgba(27, 31, 36, 0.04);
    --color-btn-inset-shadow: inset 0 1px 0 rgba(255, 255, 255, 0.25);
    color: var(--color-btn-text);
    background-color: var(--color-btn-bg);
    border-color: var(--color-btn-border);
    box-shadow: var(--color-btn-shadow), var(--color-btn-inset-shadow);
    transition: 80ms cubic-bezier(0.33, 1, 0.68, 1);
    transition-property: color, background-color, box-shadow, border-color;

    position: relative;
    display: inline-block;
    padding: 5px 16px;
    font-size: 14px;
    font-weight: 500;
    line-height: 20px;
    white-space: nowrap;
    vertical-align: middle;
    cursor: pointer;
    -webkit-user-select: none;
    user-select: none;
    border: 1px solid;
    border-radius: 6px;
    -webkit-appearance: none;
    appearance: none;

    --color-btn-hover-bg: #f3f4f6;
    --color-btn-hover-border: rgba(27, 31, 36, 0.15);
    background-color: var(--color-btn-hover-bg);
    border-color: var(--color-btn-hover-border);
}

.btn:hover,
.btn.hover,
[open]>.btn {
    --color-btn-hover-bg: #f3f4f6;
    --color-btn-hover-border: rgba(27, 31, 36, 0.15);
    background-color: var(--color-btn-hover-bg);
    border-color: var(--color-btn-hover-border);
    transition-duration: .1s;
}

.btn:active {
    --color-btn-active-bg: hsla(220, 14%, 93%, 1);
    --color-btn-active-border: rgba(27, 31, 36, 0.15);
    background-color: var(--color-btn-active-bg);
    border-color: var(--color-btn-active-border);
    transition: none;
}

@media (max-width: 481px) or (min-width: 768px) {
    .btn {
        display: none;
    }
}