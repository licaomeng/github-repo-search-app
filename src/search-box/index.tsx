import { useState } from "react";
import styles from "./SearchBox.module.css";
import { useSearchParams } from "react-router-dom";

export function SearchBox() {
    const [searchParams, setSearchParams] = useSearchParams();
    const [text, setText] = useState(searchParams.get("q") || "");

    function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
        setText(e.target.value);
    }

    function handleSubmit(e: React.FormEvent) {
        e.preventDefault();
        searchParams.set("q", text);
        setSearchParams(searchParams);
    }

    return (
        <form className={styles.form} onSubmit={handleSubmit}>
            <input
                aria-label="Search GitHub"
                autoCapitalize="off"
                autoComplete="off"
                className={styles.searchBox}
                name="q"
                maxLength={256} 
                placeholder="Search GitHub Repo"
                spellCheck="false"
                type="text"
                value={text}
                onChange={handleChange}
            ></input>
            <button className={styles.btn} type="submit">Search</button>
        </form>
    );
}