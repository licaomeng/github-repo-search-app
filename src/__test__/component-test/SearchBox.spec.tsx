import * as React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { SearchBox } from "../../components/search-box";
import { BrowserRouter as Router } from "react-router-dom";

test("renders the search-box: in header", () => {
  const { container } = render(<Router><SearchBox isInHeader={true}></SearchBox></Router>);

  expect(container).toMatchSnapshot();
});

test("renders the search-box: not in header", () => {
  const { container } = render(<Router><SearchBox isInHeader={false}></SearchBox></Router>);
  const searchBox = screen.getByTestId("search-box-submit-btn");

  expect(container).toMatchSnapshot();
  expect(searchBox).toBeVisible();
});
