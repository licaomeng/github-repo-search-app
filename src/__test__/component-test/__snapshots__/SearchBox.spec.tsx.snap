// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the search-box: in header 1`] = `
<div>
  <form
    class=""
    data-testid="search-box-form"
  >
    <input
      aria-label="Search GitHub Repo"
      autocapitalize="off"
      autocomplete="off"
      maxlength="256"
      name="q"
      placeholder="Search GitHub Repo"
      spellcheck="false"
      type="text"
      value=""
    />
  </form>
</div>
`;

exports[`renders the search-box: not in header 1`] = `
<div>
  <form
    class=""
    data-testid="search-box-form"
  >
    <input
      aria-label="Search GitHub Repo"
      autocapitalize="off"
      autocomplete="off"
      maxlength="256"
      name="q"
      placeholder="Search GitHub Repo"
      spellcheck="false"
      type="text"
      value=""
    />
    <button
      data-testid="search-box-submit-btn"
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
